src/relay/op/nn/nn.cc

```diff
@@ -259,10 +260,25 @@ bool DensePackRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   ICHECK(param != nullptr);
 
   ICHECK_EQ(data->shape.size(), 2) << "Only 2D data is supported";
-  ICHECK(weight->shape.size() == 3 || weight->shape.size() == 4) << "Expect weight to be 3D or 4D";
+  ICHECK(weight->shape.size() == 2 || weight->shape.size() == 3 || weight->shape.size() == 4) << "Expect weight to be 2D, 3D or 4D";
 
   Array<tvm::PrimExpr> oshape = data->shape;
-  oshape.Set(1, weight->shape[0] * weight->shape[2]);
+
+  std::string weight_layout = param->weight_layout;
+  std::regex blk_cn_fmt("NC[[:digit:]]+c[[:digit:]]+n");
+  std::regex blk_nc_fmt("NC[[:digit:]]+n[[:digit:]]+c");
+
+  if (weight->shape.size() == 3) {
+    oshape.Set(1, weight->shape[0] * weight->shape[2]);
```

Comment:
In this line you assume that `wgh.shape.size() == 3` means that layout is `NC_n`. I see that this logic is inherited from previous implementation. But "NC_c", "CN_c" and "CN_n" are also possible layout value.

Could you please check if assumption is correct and throw error message if layout is not "NC_n".